src/main/resources/react4xp/components/parts/MovieProcessor.ts

```diff
@@ -0,0 +1,101 @@
+import type { Content } from '@enonic-types/lib-content';
+import type { PartComponentProcessorFunction } from '@enonic-types/lib-react4xp/DataFetcher';
+import {imageUrl, pageUrl} from '/lib/xp/portal';
+import { get as getContentByKey } from '/lib/xp/content';
+
+function fetchAdditionalPhotos(photoIds: string[]) {
+    return photoIds.map(photoId => {
+        const photoContent = getContentByKey<Content>({ key: photoId });
+        return {
+            _id: photoContent._id,
+            title: photoContent.displayName,
+            imageUrl: imageUrl({ id: photoContent._id, scale: 'width(250)' }) // Image scaled for remaining photos
+        };
+    });
+}
+
+export const movieProcessor: PartComponentProcessorFunction<'com.enonic.app.hmdb:movie-details'> = (params) => {
+    const data = params.content.data;
+
+    // Extract the photos from content.data
+    const photos = data.photos;
+    const firstPhotoId = Array.isArray(photos) ? photos[0] : photos; // First photo ID
+    const remainingPhotoIds = Array.isArray(photos) ? photos.slice(1) : []; // Remaining photo IDs
```

Comment:
It's better to convert `const photos` to array from the start instead of doing it each time you use it ;)